cmake/dart_Component.cmake.in

```diff
@@ -0,0 +1,18 @@
+set("@PROJECT_NAME@_@component@_DEPENDENCIES" @internal_dependencies@)
+set("@PROJECT_NAME@_@component@_LIBRARIES" @libraries@)
+
+foreach(external_dep @external_dependencies@)
+  include(${CMAKE_CURRENT_LIST_DIR}/DARTFind${external_dep}.cmake)
+
+  string(TOUPPER ${external_dep} EXTERNAL_DEP_UPPER)
+  if(NOT ${external_dep}_FOUND AND NOT ${EXTERNAL_DEP_UPPER}_FOUND)
+    set_property(DIRECTORY PROPERTY @PROJECT_NAME_UPPERCASE@_@component@_FOUND FALSE)
+    if(NOT @PROJECT_NAME_UPPERCASE@_FIND_QUIETLY)
+      message(
+        "Cannot retrieve @PROJECT_NAME@-@component@ because the dependency "
+        "${external_dep} could not be found. This usually indicates a broken "
+        "installation.")
```

Comment:
Maybe we need to treat this as an error by specifying the mode as [`FATAL_ERROR`](https://cmake.org/cmake/help/v3.5/command/message.html)?

Reply:
No, because it might be a dependency of an optional component, which means the configuration should be allowed to pass, but that specific component should be set to `NOTFOUND`.

Reply:
Hm, I thought we assume that reaching here means the optional component was built and installed with the required optional dependencies, and failure of (re)finding dependencies here should be treated as an error. Maybe I misunderstood. In that case, how about we use `WARNING` mode instead of `(none)` (i.e., important information)?

Reply:
It's true that it's fishy if the dependency is missing, but it could certainly happen if a developer uses multiple workspace overlays and tries to configure a build with one of the overlays missing. So we should be noisy about it, but not necessarily put in a hard stop, in case they don't actually care about the optional component.

> In that case, how about we use WARNING mode instead of (none) (i.e., important information)?

That's fine; I think it's mostly an aesthetic difference in how the text will get formatted and presented to the user. If you prefer `WARNING` here, I have nothing against that.

Reply:
Should be done now.

Reply:
Thanks! :+1: